From 0c0f01b6481ba58035e1115e732cf1dadfc4dddc Mon Sep 17 00:00:00 2001
From: Fatih Uzunoglu <fuzun54@outlook.com>
Date: Tue, 8 Oct 2024 19:29:32 +0300
Subject: [PATCH] Use DirectWrite font database only with Windows 10 and above

Functionalities used by Qt are not provided with Windows
versions that are not officially supported by Qt (Windows
7, Windows 8) but supported by VLC media player.

For example, IDWriteFontFace3 is only reportedly available
starting with Windows 10. This change is necessary starting
with Qt 6.8, where DirectWrite is used by default to populate
the font database.
---
 src/plugins/platforms/windows/qwindowsintegration.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/plugins/platforms/windows/qwindowsintegration.cpp b/src/plugins/platforms/windows/qwindowsintegration.cpp
index 487e1d47b6..9e879815dc 100644
--- a/src/plugins/platforms/windows/qwindowsintegration.cpp
+++ b/src/plugins/platforms/windows/qwindowsintegration.cpp
@@ -485,7 +485,8 @@ QPlatformFontDatabase *QWindowsIntegration::fontDatabase() const
         else
 #endif // QT_NO_FREETYPE
 #if QT_CONFIG(directwrite3)
-        if (!(d->m_options & (QWindowsIntegration::FontDatabaseGDI | QWindowsIntegration::DontUseDirectWriteFonts)))
+        if (QOperatingSystemVersion::current() >= QOperatingSystemVersion::Windows10 &&
+            !(d->m_options & (QWindowsIntegration::FontDatabaseGDI | QWindowsIntegration::DontUseDirectWriteFonts)))
             d->m_fontDatabase = new QWindowsDirectWriteFontDatabase;
         else
 #endif
-- 
2.46.2

